File: output/deepseek_r1/avatar/algo_based_translation/python/java/codeforces_274_A.java

import java.util.*;

public class codeforces_274_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] firstLine = scanner.nextLine().split("\\s+");
        int num = Integer.parseInt(firstLine[0]);
        int k = Integer.parseInt(firstLine[1]);
        String[] arrLine = scanner.nextLine().split("\\s+");
        int[] arr = new int[arrLine.length];
        
        for (int i = 0; i < arrLine.length; i++) {
            arr[i] = Integer.parseInt(arrLine[i]);
        }
        
        Arrays.sort(arr);
        Set<Integer> different = new HashSet<>();
        int temp = 0;
        
        for (int x : arr) {
            if (x % k != 0 || !different.contains(x / k)) {
                different.add(x);
            }
            temp = Math.max(temp, different.size());
        }
        
        System.out.println(temp);
    }
}

File: output/llama_4_maverick/avatar/algo_based_translation/java/python/codeforces_274_A.py
n, k = map(int, input().split())
arr = list(map(int, input().split()))
arr.sort()
count = 0
unique_elements = set()
for num in arr:
    if num % k != 0 or num // k not in unique_elements:
        count += 1
        unique_elements.add(num)
print(count)
